app/models/message.rb

```diff
@@ -227,6 +228,16 @@ def save_story_info(story_info)
 
   private
 
+  def prevent_message_flooding
+    # Added this to cover the validation specs in messages
+    # We can revisit and see if we can remove this later
+    return if conversation.blank?
+
+    # there are cases where automations can result in message loops, we need to prevent such cases.
+    errors.add(:base, 'Too many messages') if conversation.messages.where('created_at >= ?',
```

Comment:
- Extend the limit to 200 per minute.
- Add Rails.logger for the error.
- Create an alert on NewRelic for this error.